src/components/buttons/ButtonIcon.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { ComponentPropsWithoutRef, forwardRef, MouseEvent, useMemo } from 'react';
+import { IconProps } from '../icons';
+import { SpinnerOverrideColor } from '../spinner/SpinnerOverrideColor';
+import { Tooltip } from '../tooltip';
+import { useButtonClickHandler } from './Button';
+import { BUTTON_VARIETY_STYLES, BUTTONICON_DIMENSIONS, ButtonIconStyled } from './ButtonStyles';
+import { ButtonSize, ButtonVariety, HTMLButtonProps } from './ButtonTypes';
+
+type TooltipProps = ComponentPropsWithoutRef<typeof Tooltip>;
+type TooltipOptions = Omit<TooltipProps, 'children' | 'content' | 'key'>;
+
+export interface ButtonIconProps extends HTMLButtonProps {
+  Icon: React.ForwardRefExoticComponent<IconProps & React.RefAttributes<HTMLSpanElement>>;
+  'aria-label': string;
+  className?: string;
+  hasAutoFocus?: boolean;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  onClick?: (event: MouseEvent<HTMLButtonElement>) => unknown;
+  size?: ButtonSize;
+  shouldPreventDefault?: boolean;
+  shouldStopPropagation?: boolean;
+  tooltipContent: TooltipProps['content'];
+  tooltipOptions?: TooltipOptions;
+  variety?: ButtonVariety;
+}
+
+export const ButtonIcon = forwardRef<HTMLButtonElement, ButtonIconProps>((props, ref) => {
+  const {
+    Icon,
+    hasAutoFocus = false,
+    isDisabled = false,
+    shouldPreventDefault = false,
+    shouldStopPropagation = false,
+    isLoading = false,
+    onClick,
+    size = ButtonSize.Large,
+    variety = ButtonVariety.Neutral,
+    tooltipContent,
```

Comment:
Should it be option and default to the ariaLabel value? I think in most case we are gonna repeat the same value as aria-label here.

Reply:
I considered it, but we often give context information in the aria-label (e.g. "edit user jean-michel" vs "edit user")

Reply:
yeah, and I think it's fine to have the same contextual information in the tooltip in most cases, no?
the point is that it avoid having to repeat the value twice when you need the same, it doesn't prevent you to pass a different value for the tooltip if you want to if it's an optional prop with a default value.